pkg/auth/authenticator/password/ldappassword/util/identityfactory.go

```diff
@@ -0,0 +1,70 @@
+package ldaputil
+
+import (
+	"fmt"
+
+	"github.com/go-ldap/ldap"
+
+	"k8s.io/kubernetes/pkg/util"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+)
+
+// LDAPUserIdentityFactory creates Identites for LDAP user entries.
+type LDAPUserIdentityFactory interface {
+	IdentityFor(user *ldap.Entry) (identity authapi.UserIdentityInfo, err error)
+}
+
+// LDAPUserToIdentityMapping holds LDAP attributes that map to OpenShift Identity fields in oder to
+// create a deterministic mapping from LDAP user entries to OpenShift Identities
+type LDAPUserToIdentityMapping struct {
```

Comment:
I don't like `LDAPEntryAttributes` because this only makes sense for a *user* entry. I also don't like `LDAPEntryAttributes` because, while this is just a collection of attribute keys,  I want the purpose of the collection to be obvious from the name.

If you don't like the name collision with other `IdentityMapping` objects, what about `LDAPUserEntryAttributeMapping`?